client/onboarding/kyc/index.tsx

```diff
@@ -15,17 +15,31 @@ import strings from 'onboarding/strings';
 import { getConnectUrl } from 'utils';
 
 const OnboardingKycPage: React.FC = () => {
+	const { detailsSubmitted: detailsSubmitted } = wcpaySettings.accountStatus;
+	const urlParams = new URLSearchParams( window.location.search );
+	const collectPayoutRequirements = !! urlParams.get(
+		'collect_payout_requirements'
+	);
+
 	const handleExit = () => {
-		const urlParams = new URLSearchParams( window.location.search );
+		const source =
+			urlParams.get( 'source' )?.replace( /[^\w-]+/g, '' ) || 'unknown';
 
-		window.location.href = getConnectUrl(
-			{
-				source:
-					urlParams.get( 'source' )?.replace( /[^\w-]+/g, '' ) ||
-					'unknown',
-			},
-			'WCPAY_ONBOARDING_KYC'
-		);
+		if ( ! detailsSubmitted ) {
+			window.location.href = getConnectUrl(
+				{
+					source: source,
+				},
+				'WCPAY_ONBOARDING_KYC'
```

Comment:
Noticed that `getConnectUrl` function arguments are the same for detailsSubmitted true and false cases
Probably a mistake here

Reply:
Fixed [here](https://github.com/Automattic/woocommerce-payments/pull/9436/commits/024aff7e3eb5eb89d059cffdf5fd54d7adc86e5e)

Reply:
Thanks Dan, I'll check it 👍

Reply:
> The intention here was to redirect to the Overview page if details are submitted (this is the case where a PO account has done initial KYC and go to the Connect page otherwise.

Oh hell... I assumed that `getConnectUrl` would retrieve the connect link provided through `wcpaySettings.connectUrl` but it doesn't; it provides the Connect page URL :( 

We  should try and use connect URLs (not Connect page URLs) - like [here](https://github.com/Automattic/woocommerce-payments/blob/024aff7e3eb5eb89d059cffdf5fd54d7adc86e5e/client/connect-account-page/index.tsx#L190-L190), and let the connect logic figure out where to point the user.

Reply:
@oaratovskyi @dmallory42 It will probably work in the current state also because we have logic to redirect from all the places :)